qa/qa-captain/bashrc

```diff
@@ -0,0 +1,1269 @@
+        # run $ go tool pprof GENERATED_FILENAME to run various commands on the files like .PNG etc
+        # trace uses $ go tool trace trace-outputfile view refresher - https://github.com/k3s-io/k3s/pull/5527 /// https://pkg.go.dev/net/http/pprof
+}
+#
+# ======================================================================================
+# ================================== ~ SONOBUOY ~ =======================================
+# ======================================================================================
+
+# --- download sonobuoy ---
+get_sono() {
+    has_bin wget
+    _arch
+    arch=$(if [ "$(uname -m)" = "x86_64" ]; then echo "amd64"; else echo "arm64"; fi)
+    wget https://github.com/vmware-tanzu/sonobuoy/releases/download/v0.56.14/sonobuoy_0.56.14_linux_"${arch}".tar.gz
+    sudo tar -xzf sonobuoy_0.56.14_linux_amd64.tar.gz -C /usr/local/bin
+}
+
+# --- start sonobuoy e2e conformance tests ---
+go_sono() {
+    _product="${1:-k3s}"
+    _version
+    _version=$(k3s --version | awk '{print $3}' | cut -c -7)
+    has_bin sonobuoy
+    #sonobuoy run --wait --kubeconfig /etc/rancher/"${product}"/"${product}".yaml --plugin https://raw.githubusercontent.com/vmware-tanzu/sonobuoy-plugins/master/cis-benchmarks/kube-bench-plugin.yaml --plugin https://raw.githubusercontent.com/vmware-tanzu/sonobuoy-plugins/master/cis-benchmarks/kube-bench-master-plugin.yaml
+    sonobuoy run --wait --kubeconfig /etc/rancher/"${_product}"/"${_product}".yaml --kubernetes-version="${_version}" --mode=certified-conformance
+}
+
+# --- view sonobuoy results ---
+sono_results() {
+    _product="${1:-k3s}"
+    has_bin sonobuoy
+    _results=$(sonobuoy retrieve --kubeconfig /etc/rancher/"${_product}"/"${_product}".yaml)
+    sonobuoy results "${_results}"    
+}
+
+# --- removes all docker images from the _cache ---
+void_docker() {
+    has_bin docker
+    docker rmi "$(docker images -a -q)"
+}
+# ======================================================================================
+# ================================== ~ RANCHER ~ =======================================
+# ======================================================================================
+
+# --- requires helm --- REWRITE WITH INSTALL.SH IF CHECKS FOR HELM --- https://github.com/k3s-io/k3s/blob/master/install.sh
+make_ranch() {
+    has_bin helm
+    has_bin kubectl
+    echo "need to adjust this to detect which binaries are already installed - modify the path based on their presences"
+    echo "figure out all the steps here to setup a _rancher node to add downstream clusters to"
+    sudo apt update && sudo apt upgrade -y
+    wait
+    helm repo add rancher-latest https://releases.rancher.com/server-charts/latest
+    kubectl create namespace cattle-system
+    helm repo add jetstack https://charts.jetstack.io
+    helm repo update
+    wait
+    helm install cert-manager jetstack/cert-manager -n cert-manager --create-namespace --version v1.5.1
```

Comment:
does this work? how do you solve the dns queries to get certs?

Reply:
Yep! Cert manager takes care of everything. This is directly from [rancher docs](https://ranchermanager.docs.rancher.com/pages-for-subheaders/install-upgrade-on-a-kubernetes-cluster#4-install-cert-manager) although we should probably update that cert manager version now. I've found `v1.8.0` works well in some previous testing.

Reply:
Updated to the later version Max referenced.